apis/codegen_test.go

```diff
@@ -0,0 +1,99 @@
+package apis
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"testing"
+
+	assert "github.com/stretchr/testify/require"
+)
+
+// TestCodeGeneration ensures that we generate the right code
+// (i.e. matches the code in services_test.go) when calling a server
+// It creates a thread that launches a server, then calls that server
+func TestCodeGeneration(t *testing.T) {
+	assert := assert.New(t)
+
+	// launch server
+	server := manifestServer()
+
+	// query server, generate code
+	source, err := GenerateServices("http://localhost:8080/manifest.json", "servicesTest", "schemasTest")
+	assert.NoError(err, fmt.Sprintf("failed generating services: %s", err))
+
+	// close server so our other tests using servers don't clash
+	server.Close()
+
+	// check that the returned byte thing is correct
+	generated := strings.Trim(string(source), "\n\r\t ")
+	desired, err := getDesiredOutput("services_test.go")
+
+	assert.NoError(err, fmt.Sprintf("error getting desired output: %s", err))
+	assert.Equal(generated, desired, "generated code doesn't match desired code")
+}
+
+// manifestServer sets up the server before launching it in a new thread
+func manifestServer() *http.Server {
+	handler := http.NewServeMux()
+	handler.HandleFunc("/manifest.json", manifestHandler)
+	handler.HandleFunc("/definition.json", apiDefHandler)
+
+	server := &http.Server{
+		Addr:    ":8080",
```

Comment:
The `:8080` address is hardcoded into **services_test.go** and **codegen_test.go**; I feel it would be a pain to have to fix everything just like that :/